public/app/features/dashboard/components/Inspector/PanelInspector.tsx

```diff
@@ -284,7 +285,7 @@ export class PanelInspectorUnconnected extends PureComponent<Props, State> {
       tabs.push({ label: 'Error', value: InspectTab.Error });
     }
 
-    if (dashboard.meta.canEdit) {
+    if (dashboard.meta.canEdit && supportsDataQuery(this.props.panel)) {
```

Comment:
Think we should use same logic as
https://github.com/grafana/grafana/blob/761b9a49d7d4cf6f95840c9c7f134d54da121a96/public/app/features/dashboard/components/Inspector/PanelInspector.tsx#L273

And use the plugin props (that is passed in via redux when it is loaded)

Reply:
> Think we should use same logic as
> https://github.com/grafana/grafana/blob/761b9a49d7d4cf6f95840c9c7f134d54da121a96/public/app/features/dashboard/components/Inspector/PanelInspector.tsx#L273
> 
> And use the plugin props (that is passed in via redux when it is loaded)

Not sure I follow what should use the same logic and use the same props?

Reply:
I think @torkelo is suggesting to pass plugin to the helper function you have created instead of the panel model?

Reply:
yes, just think we should have the same check for data tab query tab :)

Reply:
Ok, now I follow. Thanks for the clarification!